File: backend/src/ECommerce.Api/Entities/Category.cs
namespace ECommerce.Api.Entities;

public class Category
{
    public int CategoryID { get; set; }

    public string CategoryName { get; set; } = null!;

    public int? ParentID { get; set; }

    public string? Description { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    // Navigation Self-referencing
    public Category? Parent { get; set; }
    public ICollection<Category> Children { get; set; } = new List<Category>();

    // Navigation Products
    public ICollection<Product> Products { get; set; } = new List<Product>();
}

File: frontend/src/types/category.ts
export interface CategoryDto {
  categoryID: number;
  categoryName: string;
  parentID: number | null;
  description: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string | null;
}

export interface CategoryCreateDto {
  categoryName: string;
  parentID?: number | null;
  description?: string | null;
  isActive: boolean;
}

export interface CategoryUpdateDto {
  categoryName: string;
  parentID?: number | null;
  description?: string | null;
  isActive: boolean;
}
